datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -128,6 +132,15 @@ impl<T: ArrowPrimitiveType, const NULLABLE: bool> GroupColumn
         }
     }
 
+    fn append_non_nullable_val(&mut self, array: &ArrayRef, row: usize) {
+        if NULLABLE {
+            self.nulls.append(false);
```

Comment:
This could be optimized to append nulls for entire batch instead of value

Reply:
Yes, I plan to refactor the interface for supporting input a `rows: &[usize]`.
And make all parts' appending vectorized, and see the performance again.

Reply:
(i.e. remove it here and call it in such a way we use https://docs.rs/arrow/latest/arrow/array/struct.BooleanBufferBuilder.html#method.append_n

Reply:
Cool :)

Reply:
I add the `append_batch` function to support vectorized append more better.
But the improvement seems still not obvious.

🤔 I guess, it is likely due the new introduced branch of `equal_to`:
```
                if *group_idx < group_values_len {
                    for (i, group_val) in self.group_values.iter().enumerate() {
                        if !check_row_equal(group_val.as_ref(), *group_idx, &cols[i], row)
                        {
                            return false;
                        }
                    }
                } else {
                    let row_idx_offset = group_idx - group_values_len;
                    let row_idx = self.append_rows_buffer[row_idx_offset];
                    return is_rows_eq(cols, row, cols, row_idx).unwrap();
                }
```

Reply:
To eliminate this extra branch, I think we need to refactor the `intern` process metioned in https://github.com/apache/datafusion/issues/12821#issuecomment-2401542201